java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSnapshotImpl.java

```diff
@@ -318,14 +232,15 @@ protected void destroy() {
         cancel();
     }
 
-    @Override
-    public void cancel() {
+    private void cancel() {
         if (!tryRecordDisconnect()) {
             return;
         }
 
-        resultTable.handleBarrageError(new RequestCancelledException("Barrage subscription cancelled by client"));
-        GrpcUtil.safelyCancel(observer, "Barrage snapshot is cancelled", exceptionWhileCompleting);
+        final RequestCancelledException requestCancelled =
+                new RequestCancelledException("Barrage subscription cancelled by client");
+        resultTable.handleBarrageError(requestCancelled);
```

Comment:
We don't need this handleBarrageError on a future cancel, right? Nothing can get the Table in that case. And we don't really need it on a destroy, because the table had better also be dead at that point; we just need to complete the future exceptionally, if it's not already complete.

Reply:
In the snapshot case, you are right this table is not published anywhere and should be inaccessible. We do still want to do this in the subscription impl though, right? If the rpc has queued up some messages but the UG has not run the source yet, then we are able to drop those messages on the floor and do not need to rely on GC.